webapp/components/item-component/item-component.vue

```diff
@@ -2,40 +2,64 @@
   <div>
     <h2>{{ item.title }} ({{ item.votes }})</h2>
     <button class="upvote-button" @click="updateItem(item, 1)">Upvote</button>
-    <button class="downvote-button" @click="updateItem(item, -1)">Downvote</button>
-    <button class="remove-button" @click="removeItem(item)">Remove</button>
+    <button class="downvote-button" @click="updateItem(item, -1)">
+      Downvote
+    </button>
+    <button
+      class="remove-button"
+      @click="removeItem(item)"
+      v-if="loggedInAndOwner()"
+    >
+      Remove
+    </button>
+    <button class="edit-button" v-if="loggedInAndOwner()">Edit</button>
   </div>
 </template>
 
 <script>
 import Item from "../../classes/item";
+import { mapGetters, mapActions } from "vuex";
+
 export default {
   name: "ItemComponent",
   props: {
     item: Item,
   },
+  computed: {
+    ...mapGetters("auth", ["loggedIn"]),
+    ...mapGetters("auth", ["currentUserId"]),
+  },
   methods: {
+    ...mapActions('posts', ['vote']),
+    
     removeItem(item) {
       this.$emit("removeEvent", item);
     },
 
-    updateItem(item, value) {
-      const currentVotes = item.getVotes()
-      item.setVotes(currentVotes + value)
-      this.$emit("updateEvent", item);
+    async updateItem(item, value) {
+      if (this.loggedIn) {
```

Comment:
⭐ For an upvote button that behaves according to the authentication state of your user